icu4c/source/test/fuzzer/fuzzer_driver.cpp

```diff
@@ -10,11 +12,49 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size);
 
 int main(int argc, char* argv[])
 {
-  (void) argc;
-  (void) argv;
-  const char *fuzzer_data = "abc123";
-   
-  LLVMFuzzerTestOneInput((const uint8_t *) fuzzer_data, strlen(fuzzer_data));
+  bool show_warning = true;
+#if UPRV_HAS_FEATURE(address_sanitizer)
+  show_warning = false;
+#endif
+#if UPRV_HAS_FEATURE(memorys_sanitizer)
+  show_warning = false;
+#endif
+  if (argc > 2 && strcmp(argv[2], "-s") == 0) {
+    show_warning = false;
+  }
+  if (show_warning) {
+      std::cout << "WARNING: This binary work only under build configure with" << std::endl
+                << " CFLAGS=\"-fsanitize=$SANITIZE\""
+                << " CXXFLAGS=\"-fsanitize=$SANITIZE\""
+                << " ./runConfigureICU ... " << std::endl
+                << "  where $SANITIZE is address or memory" << std::endl
+                << "Plesae run the above step and make tests to rebuild" << std::endl;
+      // Do not return -1 here so we will pass the unit test.
+  }
+  if (argc < 2) {
+      std::cout << "Usage: " << argv[0] << "  testcasefile" << std::endl;
+      return -1;
+  }
+  const char *path = argv[1];
+  std::ifstream file (path, std::ios::binary );
+  if (!file.is_open()) {
+      std::cout << "Cannot open testcase file " << path << std::endl;
+      return -1;
+  }
+  file.seekg (0, std::ios::end);
+  std::streampos size = file.tellg();
+  char * memblock = new char [size];
```

Comment:
Looks like you can read the whole file into a string much more simply, something like
```
std::ostringstream ostrm;
ostrm << fin.rdbuf();
```
from https://stackoverflow.com/questions/17584784/read-a-binary-file-jpg-to-a-string-using-c